wcfsetup/install/files/lib/system/worker/UserContentRemoveWorker.class.php

```diff
@@ -175,11 +203,21 @@ public function execute() {
 		/** @var IUserContentProvider $processor */
 		$processor = ObjectTypeCache::getInstance()->getObjectType($this->data['provider'][$providerObjectType]['objectTypeID'])->getProcessor();
 		
-		$objectList = $processor->getContentListForUser($this->user);
+		$userIDs = array_keys($this->data['provider'][$providerObjectType]['user']);
+		$userID = array_shift($userIDs);
+		$user = new User($userID);
+		
+		$objectList = $processor->getContentListForUser($user);
 		$objectList->sqlLimit = $this->limit;
 		$objectList->readObjectIDs();
 		$processor->deleteContent($objectList->objectIDs);
 		
+		$this->data['provider'][$providerObjectType]['user'][$userID]['count'] -= $this->limit;
+		
+		if ($this->data['provider'][$providerObjectType]['user'][$userID]['count'] <= 0) {
+			unset($this->data['provider'][$providerObjectType]['user'][$userID]);
+		} 
+		
 		$this->data['provider'][$providerObjectType]['count'] -= $this->limit;
```

Comment:
I believe this might be buggy with the refactoring now. Consider the following:

1. A limit of 10.
2. Two users with 3 posts each – a total of 6.
3. The first user is processed: This will subtract 10 from 6, resulting in -4, resulting in the second user now being processed.

Reply:
No, because of the following code during the initialisation: https://github.com/WoltLab/WCF/blob/8ea6fb91cb5abd62470724f54c320d4df87e8ffe/wcfsetup/install/files/lib/system/worker/UserContentRemoveWorker.class.php#L170